upstairs/src/lib.rs

```diff
@@ -85,13 +88,20 @@ pub fn extent_from_offset(
     ddef: RegionDefinition,
     offset: Block,
     num_blocks: u64,
-) -> Result<Vec<(u64, Block, u64)>> {
-    assert!(num_blocks > 0);
-    assert!(
-        (offset.value + num_blocks)
-            <= (ddef.extent_size().value * ddef.extent_count() as u64)
-    );
-    assert_eq!(offset.block_size_in_bytes() as u64, ddef.block_size());
+) -> Result<Vec<(u64, Block, u64)>, CrucibleError> {
+    if num_blocks == 0 {
+        crucible_bail!(InvalidNumberOfBlocks, "num_blocks == 0!");
```

Comment:
I imagined us collecting this error and maybe some request metadata, hopefully to aid debugging how that bail happened in the first place, instead of blowing up us and the VM.